src/Symfony/Component/Form/Extension/Core/Type/ChoiceType.php

```diff
@@ -391,14 +405,24 @@ private function createChoiceList(array $options)
     {
         if (null !== $options['choice_loader']) {
             return $this->choiceListFactory->createListFromLoader(
-                $options['choice_loader'],
+                null === $options['choice_filter'] ? $options['choice_loader'] : new FilterChoiceLoader($options['choice_loader'], $options['choice_filter']),
                 $options['choice_value']
             );
         }
 
         // Harden against NULL values (like in EntityType and ModelType)
         $choices = null !== $options['choices'] ? $options['choices'] : array();
 
```

Comment:
what i like to do here is:

```php
$loader = new CallbackChoiceLoader(function () use ($choices) {
    return $choices;
});
```

and then proceed as above (like `choice_loader`)